pom.xml

```diff
@@ -1192,18 +1187,28 @@
                     <release>11</release>>
                 </configuration>
             </plugin>
+            <!--Resolve EclipseLink Parent project root directory-->
             <plugin>
-                <groupId>org.commonjava.maven.plugins</groupId>
-                <artifactId>directory-maven-plugin</artifactId>
+                <groupId>org.codehaus.gmaven</groupId>
+                <artifactId>groovy-maven-plugin</artifactId>
                 <executions>
                     <execution>
                         <id>find-project-root</id>
                         <phase>validate</phase>
                         <goals>
-                            <goal>highest-basedir</goal>
+                            <goal>execute</goal>
                         </goals>
                         <configuration>
-                            <property>project.root.location</property>
+                            <source>
+                                <![CDATA[
+                                    def rootPom = project;
+                                    while (rootPom.parent.groupId != "org.eclipse.ee4j") {
+                                        rootPom = rootPom.parent;
+                                    }
+                                    project.properties.setProperty("project.root.location", rootPom.basedir.toString());
+                                    log.info("EclipseLink Parent project directory [project.root.location]:  " + project.properties['project.root.location']);
+                                ]]>
```

Comment:
Why can't we rely on `${maven.multiModuleProjectDirectory}` or `${session.executionRootDirectory}`? Do we really support running `mvn` in submodules (ie without `mvn -pl :...`)?

Reply:
You cannot rely on `${maven.multiModuleProjectDirectory}`. This is private property (implementation detail).

Reply:
What else would you recommend?

Reply:
You basically need to solve: https://issues.apache.org/jira/browse/MNG-7038

Reply:
to keep resources shared by all submodules on one place without a need to define a to be redefined in each submodule property, to avoid having multiple copies of the same file on multiple places through the repo or to avoid having to maintain an extra artifact having them. Examples of such resources: license/notice files from the repo root which every build archive must include or common configuration files for various plugins (can be copyright template, common spotbugs exclusion filter, etc)

Reply:
You name it: avoid having to maintain an extra artifact having them. That is the usecase for a POM module with attached classified artifact. @rfscholte or do you see a better way?

Reply:
A shared resource doesn't have to be restricted classes, it can be of any type. So the preferred way is indeed a artifact that can be used as dependency. Whether it is a classified artifact of "root" or a new modX, that's up to the user.
I still aim for proper context: Maven modules should never look outside their own directory.

Reply:
right, shared resource can be anything. Not sure if all plugins support reading files from such resource, so it may be required to add additional copy/unpack step into submodules to put them into project.build.directory during the build unless one wants to fix all such plugins...

Reply:
No they don't. You have already described the correct process. This is what we do in Maven and me in my personal OSS projects.